### PR TITLE
removes detectives baton and gives them a flash

### PR DESCRIPTION
**Pull request by boodaliboo**
they arent security they dont need it and if they need to stun this pr gives them a flash
#### Changelog

:cl:  
rscdel: Removed detectives baton 
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/detective.dm [hunk 1]
```diff
@@ -49,8 +49,7 @@
 	l_pocket = /obj/item/toy/crayon/white
 	r_pocket = /obj/item/lighter
 	backpack_contents = list(/obj/item/storage/box/evidence=1,\
-		/obj/item/detective_scanner=1,\
-		/obj/item/melee/classic_baton=1)
+		/obj/item/detective_scanner=1)
```
Comment (ynot01) [inline code/modules/jobs/job_types/detective.dm]:
```suggestion
		/obj/item/detective_scanner=1,\
		/obj/item/assembly/flash)
		
```
give them a flash instead and ill support you

**Comment by Marmio64** [comment]
this is actually really fucking great

detective doesn't need a baton to valid antags, the gun is for self defense only.

**Comment by boodaliboo** [comment]
done ynot

**Comment by Tipy1802** [comment]
The detective is actually permitted to arrest if they see the crime happen. They also need it to detain for questioning

**Comment by Tipy1802** [comment]
Also if you will remove their baton remove their mindshield so they aren't taken out of the round every time there is a convertion antag since they wouldn't be able to defend themself very well. Sure they have a gun but it's crap and it takes multiple hits to stun someone, while gangs have proper guns, cultists stun spells and revs often use stunprods and stolen stun batons

**Comment by Tipy1802** [comment]
Another reason the detective needs a batton is because they are supposed to act like an officer if security is understaffed or an emergency forces them to, which happens rather often

**Comment by mkikt4743** [comment]
ok, I'll just use crit as a stun then

**Comment by boba247** [comment]
By this logic of only sec should have batons then heads shouldn’t have batons either. Except detectives are supposed to intervene in emergencies something heads aren’t required to do.

**Comment by Tipy1802** [comment]
All this PR does is make the detective ask warden/hos/captain for a proper stun baton

**Comment by JamieD1** [comment]
Sppaaaaaaaaaacccceeee

**Comment by Marmio64** [comment]
> All this PR does is make the detective ask warden/hos/captain for a proper stun baton

thats not even allowed



> By this logic of only sec should have batons then heads shouldn’t have batons either. Except detectives are supposed to intervene in emergencies something heads aren’t required to do.

it would actually be good if heads of staff didnt get telescopic batons at all and instead only had flashes, the batons encourage them to become hopcurity/cmocurity/etc when they really shouldnt.

**Comment by boba247** [comment]
>  the batons encourage them to become hopcurity/cmocurity/etc when they really shouldnt.

Except we have rules against this. The HOP even has a specific rule against involving themselves with sec. 
Code-based changes to prevent valid hunting do work, the sec borg change is a prime example. However for these changes to be necessary there needs to be a serious problem of valid hunting. Removing the detective's baton won't stop detective's that want to overstep their bounds from overstepping their bounds. To stop valid hunting with the detective you need a major rework of the detective into something like a civilian forensic tech.

**Comment by Tipy1802** [comment]
> > All this PR does is make the detective ask warden/hos/captain for a proper stun baton
> 
> thats not even allowed
> 
It is allowed. As detective I have asked wardens for sec HUDs and I have recived them. I have had HoS give me flash bangs and wardens give me disablers without me even asking.

Yogs' detective is already so much worse than the detective of most servers and this only makes him worse. Other servers have things like autopsies, blood trackers and way more tools to solve crime and track criminals down. What does yogs' detective have? The forensic scanner and just that. The forensic scanner isn't that fun to use either, just click on something you belive may have been touched by the criminal and cry when you get ten billion prints and fibers of different people and jobs or just get everyone's favourite ''fibers from insulated gloves'' Thanks god it's not like 60% of the station wears insulated gloves. All it takes to fuck the detective is just changing clothing before commiting a crime.

If you would like to improve the detective I can link you the wiki pages for the detective in other servers for inspiration:
https://wiki.ss13.co/Detective
https://wiki.aurorastation.org/index.php?title=Forensic_Technician

**Comment by boodaliboo** [comment]
i dont care

**Comment by alexkar598** [comment]
I think reducing the stun on the baton to ensure its a self defense weapon that cant be used to cuff someone would be a better choice here

**Comment by MegaEmpirical** [comment]
Detectives are security.

**Pull request closed without merging**